python/python/bystro/api/proteomics.py

```diff
@@ -0,0 +1,185 @@
+    update the annotation job.
+
+    Parameters
+    ----------
+    protein_abundance_file : str
+        Path to the protein abundance file.
+    experiment_annotation_file : str | None
+        Path to the experiment annotation file.
+    annotation_job_id : str | None
+        annotationID of the job associated with the annotation dataset.
+    proteomics_dataset_type : DatasetTypes
+        Type of the proteomics dataset (we only support fragpipe-TMT currently).
+    experiment_name : str | None
+        Name of the experiment, required if the experiment annotation file contains multiple experiments
+    print_result : bool
+        Whether to print the result of the upload operation, by default True.
+
+    Returns
+    -------
+    dict
+        A json response with annotationID and proteomicsID.
+    """
+
+    if annotation_job_id is not None and not isinstance(annotation_job_id, str):
+        raise ValueError("annotation job id must be a string.")
+
+    with open(protein_abundance_file, "r") as file:
+        first_line = file.readline().strip().lower()
+        if not all(header.lower() in first_line for header in FRAGPIPE_GENE_ABUNDANCE_HEADERS):
+            raise ValueError(
+                "The protein abundance file does not contain the expected headers: %s"
+                % FRAGPIPE_GENE_ABUNDANCE_HEADERS
+            )
+
+    if experiment_annotation_file:
+        with open(experiment_annotation_file, "r") as file:
+            first_line = file.readline().strip().lower()
+            if not all(header.lower() in first_line for header in EXPERIMENT_ANNOTATION_HEADERS):
+                raise ValueError(
+                    "The experiment annotation file does not contain the expected headers: %s"
+                    % EXPERIMENT_ANNOTATION_HEADERS
+                )
+
+            experiment_name_index = EXPERIMENT_ANNOTATION_HEADERS.index("Experiment Name")
+            experiment_names = set()
+            for line in file:
+                row = line.strip().split("\t")
+                experiment_names.add(row[experiment_name_index])
+
+        if len(experiment_names) == 1:
+            experiment_name = experiment_name or experiment_names.pop()
+        elif not experiment_name:
+            raise ValueError(
+                "Either no experiment name or multiple experiment names "
+                "were found in the experiment annotation file. "
+                "Please provide an experiment name."
+            )
+
+        state, auth_header = authenticate()
+        experiment_file = _package_filename(experiment_annotation_file)
+
+        experiment_response = requests.post(
+            state.url + EXPERIMENT_ENDPOINT,
+            headers=auth_header,
+            files={"file": experiment_file[1]},
+            data={"experimentName": experiment_name},
+        )
+        if experiment_response.status_code != HTTP_STATUS_OK:
+            raise Exception(
```

Comment:
Please convert this to a RuntimeError, squash this PR, and then we can get it in.